flink-python/pyflink/table/catalog.py

```diff
@@ -0,0 +1,909 @@
+        if j_catalog_table_statistics is None:
+            if properties is None:
+                self._j_catalog_table_statistics = gateway.jvm.CatalogTableStatistics(
+                    row_count, field_count, total_size, raw_data_size)
+            else:
+                self._j_catalog_table_statistics = gateway.jvm.CatalogTableStatistics(
+                    row_count, field_count, total_size, raw_data_size, properties)
+        else:
+            self._j_catalog_table_statistics = j_catalog_table_statistics
+
+    def get_row_count(self):
+        """
+        The number of rows in the table or partition.
+        """
+        return self._j_catalog_table_statistics.getRowCount()
+
+    def get_field_count(self):
+        """
+        The number of files on disk.
+        """
+        return self._j_catalog_table_statistics.getFileCount()
+
+    def get_total_size(self):
+        """
+        The total size in bytes.
+        """
+        return self._j_catalog_table_statistics.getTotalSize()
+
+    def get_raw_data_size(self):
+        """
+        The raw data size (size when loaded in memory) in bytes.
+        """
+        return self._j_catalog_table_statistics.getRawDataSize()
+
+    def get_properties(self):
+        return dict(self._j_catalog_table_statistics.getProperties())
+
+    def copy(self):
+        """
+        Create a deep copy of "this" instance.
+        """
+        return CatalogTableStatistics(
+            j_catalog_table_statistics=self._j_catalog_table_statistics.copy())
+
+
+class CatalogColumnStatistics(object):
+    """
+    Column statistics of a table or partition.
+    """
+
+    def __init__(self, column_statistics_data=None, properties=None,
+                 j_catalog_column_statistics=None):
+        if j_catalog_column_statistics is None:
+            gateway = get_gateway()
+            java_import(gateway.jvm, "org.apache.flink.table.catalog.stats.CatalogColumnStatistics")
+            if properties is None:
+                self._j_catalog_column_statistics = gateway.jvm.CatalogColumnStatistics(
+                    column_statistics_data)
+            else:
+                self._j_catalog_column_statistics = gateway.jvm.CatalogColumnStatistics(
+                    column_statistics_data, properties)
+        else:
+            self._j_catalog_column_statistics = j_catalog_column_statistics
+
+    def get_column_statistics_data(self):
+        return self._j_catalog_column_statistics.getColumnStatisticsData()
+
+    def get_properties(self):
+        return dict(self._j_catalog_column_statistics.getProperties())
+
+    def copy(self):
+        return CatalogColumnStatistics(
+            j_catalog_column_statistics=self._j_catalog_column_statistics.copy())
+
+
+class HiveCatalog(Catalog):
+    """
+    A catalog implementation for Hive.
+    """
+
+    def __init__(self, catalog_name, default_database="default", hive_metastore_uri=None,
```

Comment:
we recently changed the way to instantiate a HiveCatalog, pls checkout its latest constructor signature

Reply:
Good catch. +1